dot/rpc/modules/system.go

```diff
@@ -135,3 +151,47 @@ func (sm *SystemModule) NodeRoles(r *http.Request, req *EmptyRequest, res *[]int
 	*res = resultArray
 	return nil
 }
+
+// AccountNextIndex Returns the next valid index (aka. nonce) for given account.
+func (sm *SystemModule) AccountNextIndex(r *http.Request, req *StringRequest, res *U64Response) error {
+	// todo (ed) check pending transactions
+
+	// get metadata to build storage storageKey
+	rawMeta, err := sm.coreAPI.GetMetadata(nil)
+	if err != nil {
+		return err
+	}
+	sdMeta, err := scale.Decode(rawMeta, []byte{})
+	if err != nil {
+		return err
+	}
+	var metadata types.Metadata
+	err = types.DecodeFromBytes(sdMeta.([]byte), &metadata)
+	if err != nil {
+		return err
+	}
+
+	if req == nil || len(req.String) == 0 {
+		return errors.New("Account address must be valid")
```

Comment:
maybe move this to the top of the function so it checks it first?

Reply:
good idea.